posthog/api/feature_flag.py

```diff
@@ -252,6 +252,40 @@ def local_evaluation(self, request: request.Request, **kwargs):
             }
         )
 
+    @action(methods=["GET"], detail=False)
+    def evaluation_reasons(self, request: request.Request, **kwargs):
+
+        distinct_id = request.query_params.get("distinct_id", None)
+        groups = json.loads(request.query_params.get("groups", "{}"))
+
+        if not distinct_id:
+            raise exceptions.ValidationError(detail="distinct_id is required")
+
+        flags, reasons = get_active_feature_flags(self.team_id, distinct_id, groups)
+
+        flags_with_evaluation_reasons = {}
+
+        for flag_key in reasons:
+            flags_with_evaluation_reasons[flag_key] = {
```

Comment:
does it make sense to also say whether a feature flag is part of an experiment or not?

Reply:
(asking because we show this in the figma mocks)

Reply:
You'll get that info from the `featureflagmodel` / `featureflaglistlogic` ?

Reply:
sure but then we'll need to do a mapping to it to determine the type and ID when we can just grab it straight from the evaluation_reasons endpoint?

Reply:
Could do it, but doesn't make much sense to me right now, since `evaluation_reasons` right now doesn't have this info. (it only gets keys, values and evaluation reasons).

So, the only difference will be doing the mapping in the backend vs the frontend. But, the frontend already has the other flags data, so seems better to do it there, and keeps this endpoint clean for clients to call directly.